xmodule/library_content_block.py

```diff
@@ -159,12 +155,29 @@ def completion_mode(cls):  # pylint: disable=no-self-argument
         values=_get_capa_types(),
         scope=Scope.settings,
     )
+    candidates = List(
+        # This is a list of (block_type, block_id) used to record the set of blocks
+        # which are candidates for the selected list.
+        display_name=_("Manually Selected Blocks"),
+        default=[],
+        scope=Scope.settings,
+    )
     selected = List(
         # This is a list of (block_type, block_id) tuples used to record
-        # which random/first set of matching blocks was selected per user
+        # which set of matching blocks was selected per user
         default=[],
         scope=Scope.user_state,
     )
+    shuffle = Boolean(
+        # Are the blocks seen and their order randomized for each learner?
+        default=True,
+        scope=Scope.settings,
+    )
+    manual = Boolean(
+        #is the block content only drawn from content which is in the candidates list?
+        default=False,
+        cope=Scope.settings,
+    )
```

Comment:
> The interactive editor you and Ray are working on will be much nicer. I think that's how we ought to debut the static ref capability to general users. Do you agree?

The nascent plan I had was to merge this after we are ready to launch the new editor.

> If so, then for this PR, we could declare the new settings as hidden=True.

iff that doesn't hide them from the xblock api, then yes (I will check that).

> I think the JS supporting the legacy editor could go. Would the reordering support on the children page also go away, or is that still necessary even with the new editor?

That JS supports the "view library children" page, which we are not removing, because the new editor does not support the editing of children. Therefore it is necessary, for now.

Reply:
@connorhaugh 

> The nascent plan I had was to merge this after we are ready to launch the new editor. I realize that we should probably make the old editor not notice these changes, in order to preserve the experience in the event that we have to turn off the new library block editor. Thanks for bringing up that point.
> iff that doesn't hide them from the xblock api, then yes (I will check that).

👍🏻 

> That JS supports the "view library children" page, which we are not removing, because the new editor does not support the editing of children. Therefore it is necessary, for now.

I understand we need to keep the "view library children" page, and that we need to make that page aware of `.candidates`, but I think the JS/HTML changes in this PR go beyond just that:

* library_content_block.js seems to support the legacy block picker, which will be unused.
* The edit to container.html which adds a "Save" button is there to support re-ordering, right? Do we want to support re-ordering now, or will the new component editor handle that?

Reply:
> The edit to container.html which adds a "Save" button is there to support re-ordering, right?

No, it is there to support candidate selection on that page.

Reply:
> library_content_block.js seems to support the legacy block picker, which will be unused.
i'll get rid of this stuff.

Reply:
excellent 👍🏻